serve/plugin.go

```diff
@@ -1,67 +1,109 @@
 
-	serversv2 "github.com/cloudquery/plugin-sdk/v3/internal/servers/source/v2"
-	"github.com/cloudquery/plugin-sdk/v3/plugins/source"
+	serverDestinationV0 "github.com/cloudquery/plugin-sdk/v4/internal/servers/destination/v0"
+	serverDestinationV1 "github.com/cloudquery/plugin-sdk/v4/internal/servers/destination/v1"
+	serversv3 "github.com/cloudquery/plugin-sdk/v4/internal/servers/plugin/v3"
 	"github.com/getsentry/sentry-go"
 	grpczerolog "github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 	"github.com/thoas/go-funk"
-	"golang.org/x/net/netutil"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/test/bufconn"
 )
 
-type sourceServe struct {
-	plugin    *source.Plugin
-	sentryDSN string
+type PluginServe struct {
+	plugin                *plugin.Plugin
+	args                  []string
+	destinationV0V1Server bool
+	sentryDSN             string
+	testListener          bool
+	testListenerConn      *bufconn.Listener
 }
 
-type SourceOption func(*sourceServe)
+type PluginOption func(*PluginServe)
 
-func WithSourceSentryDSN(dsn string) SourceOption {
-	return func(s *sourceServe) {
+func WithPluginSentryDSN(dsn string) PluginOption {
+	return func(s *PluginServe) {
 		s.sentryDSN = dsn
 	}
 }
 
-// lis used for unit testing grpc server and client
-var testSourceListener *bufconn.Listener
-var testSourceListenerLock sync.Mutex
+// WithDestinationV0V1Server is used to include destination v0 and v1 server to work
+// with older sources
+func WithDestinationV0V1Server() PluginOption {
+	return func(s *PluginServe) {
+		s.destinationV0V1Server = true
+	}
+}
+
+// WithArgs used to serve the plugin with predefined args instead of os.Args
+func WithArgs(args ...string) PluginOption {
+	return func(s *PluginServe) {
+		s.args = args
+	}
+}
+
+// WithTestListener means that the plugin will be served with an in-memory listener
+// available via testListener() method instead of a network listener.
+func WithTestListener() PluginOption {
+	return func(s *PluginServe) {
+		s.testListener = true
+		s.testListenerConn = bufconn.Listen(testBufSize)
+	}
+}
 
-const serveSourceShort = `Start source plugin server`
+const servePluginShort = `Start plugin server`
 
-func Source(plugin *source.Plugin, opts ...SourceOption) {
-	s := &sourceServe{
-		plugin: plugin,
+func Plugin(p *plugin.Plugin, opts ...PluginOption) *PluginServe {
+	s := &PluginServe{
+		plugin: p,
 	}
 	for _, opt := range opts {
 		opt(s)
 	}
-	if err := newCmdSourceRoot(s).Execute(); err != nil {
-		sentry.CaptureMessage(err.Error())
-		fmt.Println(err)
-		os.Exit(1)
+	return s
+}
+
+func (s *PluginServe) bufPluginDialer(context.Context, string) (net.Conn, error) {
+	return s.testListenerConn.Dial()
+}
+
+func (s *PluginServe) Serve(ctx context.Context) error {
```

Comment:
The returned `error` here is not used (or is a hassle to use) from a `main.go` POV. Maybe make a `ServeConsole()`  helper func that calls Serve with context.Background(), doesn't return error, but prints error to stderr (and exits with proper error code) instead? I know it's